src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -159,6 +173,424 @@ struct kds_cu_info
+      {"Stream_Switch_Slave_West_3_Slot0", 0x0003F300},
+      {"Stream_Switch_Slave_West_3_Slot1", 0x0003F304},
+      {"Stream_Switch_Slave_West_3_Slot2", 0x0003F308},
+      {"Stream_Switch_Slave_West_3_Slot3", 0x0003F30C},
+      {"Stream_Switch_Slave_North_0_Slot0", 0x0003F310},
+      {"Stream_Switch_Slave_North_0_Slot1", 0x0003F314},
+      {"Stream_Switch_Slave_North_0_Slot2", 0x0003F318},
+      {"Stream_Switch_Slave_North_0_Slot3", 0x0003F31C},
+      {"Stream_Switch_Slave_North_1_Slot0", 0x0003F320},
+      {"Stream_Switch_Slave_North_1_Slot1", 0x0003F324},
+      {"Stream_Switch_Slave_North_1_Slot2", 0x0003F328},
+      {"Stream_Switch_Slave_North_1_Slot3", 0x0003F32C},
+      {"Stream_Switch_Slave_North_2_Slot0", 0x0003F330},
+      {"Stream_Switch_Slave_North_2_Slot1", 0x0003F334},
+      {"Stream_Switch_Slave_North_2_Slot2", 0x0003F338},
+      {"Stream_Switch_Slave_North_2_Slot3", 0x0003F33C},
+      {"Stream_Switch_Slave_North_3_Slot0", 0x0003F340},
+      {"Stream_Switch_Slave_North_3_Slot1", 0x0003F344},
+      {"Stream_Switch_Slave_North_3_Slot2", 0x0003F348},
+      {"Stream_Switch_Slave_North_3_Slot3", 0x0003F34C},
+      {"Stream_Switch_Slave_East_0_Slot0", 0x0003F350},
+      {"Stream_Switch_Slave_East_0_Slot1", 0x0003F354},
+      {"Stream_Switch_Slave_East_0_Slot2", 0x0003F358},
+      {"Stream_Switch_Slave_East_0_Slot3", 0x0003F35C},
+      {"Stream_Switch_Slave_East_1_Slot0", 0x0003F360},
+      {"Stream_Switch_Slave_East_1_Slot1", 0x0003F364},
+      {"Stream_Switch_Slave_East_1_Slot2", 0x0003F368},
+      {"Stream_Switch_Slave_East_1_Slot3", 0x0003F36C},
+      {"Stream_Switch_Slave_East_2_Slot0", 0x0003F370},
+      {"Stream_Switch_Slave_East_2_Slot1", 0x0003F374},
+      {"Stream_Switch_Slave_East_2_Slot2", 0x0003F378},
+      {"Stream_Switch_Slave_East_2_Slot3", 0x0003F37C},
+      {"Stream_Switch_Slave_East_3_Slot0", 0x0003F380},
+      {"Stream_Switch_Slave_East_3_Slot1", 0x0003F384},
+      {"Stream_Switch_Slave_East_3_Slot2", 0x0003F388},
+      {"Stream_Switch_Slave_East_3_Slot3", 0x0003F38C},
+      {"Stream_Switch_Slave_ME_Trace_Slot0", 0x0003F390},
+      {"Stream_Switch_Slave_ME_Trace_Slot1", 0x0003F394},
+      {"Stream_Switch_Slave_ME_Trace_Slot2", 0x0003F398},
+      {"Stream_Switch_Slave_ME_Trace_Slot3", 0x0003F39C},
+      {"Stream_Switch_Slave_Mem_Trace_Slot0", 0x0003F3A0},
+      {"Stream_Switch_Slave_Mem_Trace_Slot1", 0x0003F3A4},
+      {"Stream_Switch_Slave_Mem_Trace_Slot2", 0x0003F3A8},
+      {"Stream_Switch_Slave_Mem_Trace_Slot3", 0x0003F3AC},
+      {"Stream_Switch_Event_Port_Selection_0", 0x0003FF00},
+      {"Stream_Switch_Event_Port_Selection_1", 0x0003FF04} };
+ 
+
+  std::string err;
+  std::string value;
+  dev->sysfs_get("aie_metadata", err, value);
+  if (!err.empty())
+    throw xrt_core::error(-EINVAL, err);
+  std::stringstream ss(value);
+  boost::property_tree::ptree pt;
+  boost::property_tree::read_json(ss, pt);
+   
+  int mKernelFD = open("/dev/dri/renderD128", O_RDWR);
```

Comment:
Magic strings....please define this value in a `const static string`.

Reply:
fixed